Menus/UndoMenu/Scripts/UndoMenuUI.cs

```diff
@@ -0,0 +1,175 @@
+        Coroutine m_UndoPerformedCoroutine;
+
+        public Transform alternateMenuOrigin
+        {
+            get { return m_AlternateMenuOrigin; }
+            set
+            {
+                if (m_AlternateMenuOrigin == value)
+                    return;
+
+                m_AlternateMenuOrigin = value;
+                transform.SetParent(m_AlternateMenuOrigin);
+                transform.localPosition = Vector3.zero;
+                transform.localRotation = Quaternion.identity;
+            }
+        }
+
+        Transform m_AlternateMenuOrigin;
+
+        public bool engaged
+        {
+            get { return m_Engaged; }
+            set
+            {
+                if (m_Engaged == value)
+                    return;
+                m_Engaged = value;
+                this.RestartCoroutine(ref m_EngageCoroutine, AnimateEngage(m_Engaged));
+            }
+        }
+
+        bool m_Engaged;
+
+        public bool visible
+        {
+            get { return m_Visible; }
+            set
+            {
+                if (m_Visible == value)
+                    return;
+
+                m_Visible = value;
+
+                gameObject.SetActive(value);
+            }
+        }
+
+        bool m_Visible;
+
+        public List<ActionMenuData> actions
+        {
+            get { return m_Actions; }
+            set
+            {
+                if (value != null)
+                {
+                    m_Actions = value
+                        .Where(a => a.sectionName != null && a.sectionName == ActionMenuItemAttribute.DefaultActionSectionName)
+                        .OrderBy(a => a.priority)
+                        .ToList();
+                }
+                else if (visible)
+                    visible = false;
+            }
+        }
+
+        List<ActionMenuData> m_Actions;
+
+        void Awake()
+        {
+            m_UndoButtonMaterial = MaterialUtils.GetMaterialClone(m_UndoButtonMeshRenderer);
+            m_RedoButtonMaterial = MaterialUtils.GetMaterialClone(m_RedoButtonMeshRenderer);
+        }
+
+        void OnDestroy()
+        {
+            ObjectUtils.Destroy(m_UndoButtonMaterial);
+            ObjectUtils.Destroy(m_RedoButtonMaterial);
+        }
+
+        public void Setup()
+        {
+            gameObject.SetActive(false);
+        }
+
+        IEnumerator AnimateEngage(bool engaging)
+        {
+            var undoStartingColor = m_UndoButtonMaterial.GetColor(k_MaterialColorProperty);
+            var redoStartingColor = m_RedoButtonMaterial.GetColor(k_MaterialColorProperty);
+            var targetColor = Color.white;
+            targetColor.a = engaging ? k_EngagedAlpha : k_DisengagedAlpha;
+            var transitionAmount = 0f;
+            var currentDuration = 0f;
+            while (transitionAmount < 1f)
+            {
+                m_UndoButtonMaterial.SetColor(k_MaterialColorProperty, Color.Lerp(undoStartingColor, targetColor, transitionAmount));
```

Comment:
Use SmoothInOutLerpFloat()
var smoothAmount = MathUtilsExt.SmoothInOutLerpFloat(transitionAmount);
                m_UndoButtonMaterial.SetColor(k_MaterialColorProperty, Color.Lerp(undoStartingColor, targetColor, smoothAmount));
                m_RedoButtonMaterial.SetColor(k_MaterialColorProperty, Color.Lerp(redoStartingColor, targetColor, smoothAmount));

Reply:
Orrrr maybe AnimateProperty, once that lands? I'm inclined to punt for that reason.

Reply:
I've switched to SmoothInOutLerp for now, but yep, that AnimateProperty sounds pretty good. Where's that branchwise?

Reply:
It's in Dev now. https://github.com/Unity-Technologies/EditorVR/blob/development/Scripts/Utilities/TransitionUtils.cs

Here's how it's used https://github.com/Unity-Technologies/EditorVR/blob/development/Scripts/Proxies/ProxyNode.cs#L146